sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -282,34 +282,34 @@ object FunctionRegistry {
 
     // aggregate functions
     expression[HyperLogLogPlusPlus]("approx_count_distinct"),
-    expression[Average]("avg"),
+    expressionWithAlias[Average]("avg"),
+    expressionWithAlias[Average]("mean"),
```

Comment:
Seems we don't need `expressionWithAlias` at all, just call  `expression[Average]("mean", setAlias = true)`.

And `expression[Average]("avg")` can remain unchanged, as `avg` is already the default name.

Reply:
And we don't need to reorder them now.